src/cc/libbpf.c

```diff
@@ -156,6 +160,13 @@ int bpf_prog_load(enum bpf_prog_type prog_type,
     log_buf[0] = 0;
 
   int ret = syscall(__NR_bpf, BPF_PROG_LOAD, &attr, sizeof(attr));
+  if (attr.insn_cnt > BPF_MAXINSNS) {
```

Comment:
I like the fact that this error case is checked, but it is problematic when done after the syscall. Can you move the check to be above the syscall?